technical-doc/bootstrap.rst

```diff
@@ -0,0 +1,137 @@
+than new blocks appear in the meantime, and would take a very long time.
+Moreover, Massa aims at processing thousands of operations per second,
+which means that it produces a lot of block data every second,
+thus preventing nodes with the target hardware from storing the full block history
+and making bootstrapping from genesis impossible since old blocks are forgotten.
+
+Massa nodes joining the network must therefore bootstrap by downloading the absolute current State.
+
+Note that nodes can recover from short-term disconnects by asking for missing data from nodes around them once they come back in the network.
+However, since Massa nodes only store a short history of blocks and forget older ones, it is impossible to recover from long disconnects
+since the surrounding nodes have forgotten the blocks needed by the recovering node. In that case, a new State bootstrap is required.
+
+
+Security model
+==============
+
+The example of Bitcoin
+----------------------
+
+To understand the security model of node bootstrap, Bitcoin is a good starting example.
+
+When Bitcoin node runners decide to join the network, they first download the node software from a central source (eg. bitcoin.org).
+If that source is compromised, the node might end up on a different network, and/or private key theft might happen.
+Bitcoin therefore requires trust in the entity sourcing the node software.
+
+Note that optionally, if the nodes don't wish to download the full block history
+they resort to trusting a more recent "checkpoint" state encoded in the node software.
+If that state is compromised, the node's knowledge of the whole ledger might be skewed.
+
+Assuming the node software is not corrupted, new nodes joining the network need an initial list of peer nodes
+to which they need to connect first in order to discover the network.
+This list is hardcoded in the node software (see https://developer.bitcoin.org/devguide/p2p_network.html#peer-discovery).
+If all peers pointed by this initial list are compromised, the node can end on a different (non-bitcoin) network,
+even if the node software itself is not compromised.
+Bitcoin therefore also requires trust in at least one of the initial peers.
+
+Note that ending up on the wrong network can be detected by checking block hashes with an external source.
+But this requires trusting yet another source of data.
+
+The case of Massa
+-----------------
+
+The Massa case is very similar to Bitcoin's.
+Node runners also need to trust the source of the node software they download,
+as well as the initial list of peers.
+
+Similarly to Bitcoin checkpoints, bootstrapping Massa nodes must obtain the current State from a trusted source,
+ideally the same source as the one they downloaded the node software from, in order to avoid having to trust multiple entities.
+
+Downloading the state from an untrusted source can result in major issues such as coin theft.
```

Comment:
Maybe we could give more examples, arguments on why we implement the whitelist (that strongly force people to bootstrap from our nodes)

Reply:
I added a paragraph